symplyphysics/core/experimental/coordinate_systems/base_system.py

```diff
@@ -43,32 +43,36 @@ def __repr__(self) -> str:
 
     def __new__(
         cls,
-        base_scalars: Optional[tuple[Symbol, Symbol, Symbol]] = None,
-        base_scalar_functions: Optional[tuple[Function, Function, Function]] = None,
+        base_scalars: Optional[Sequence[Symbol]] = None,
+        base_scalar_functions: Optional[Sequence[Symbol]] = None,
     ) -> BaseCoordinateSystem:
         if base_scalars:
-            a, b, c = tuple(base_scalars)
-            base_scalars = a, b, c
+            a, b, c = base_scalars
+            base_scalars_ = a, b, c
+        else:
+            base_scalars_ = None
```

Comment:
Not the best idea to define variable in the nested block. Python allows it but most other languages forbid it.
Instead define `base_scalars_` outside.